sky/packages/sky/lib/src/widgets/scrollable.dart

```diff
@@ -379,6 +407,44 @@ class Block extends StatelessComponent {
   }
 }
 
+abstract class ScrollableListPainter extends Painter {
+  void attach(RenderObject renderObject) {
+    assert(renderObject is RenderBlockViewport);
+    super.attach(renderObject);
+  }
+
+  RenderBlockViewport get renderer => renderObject;
+
+  bool get isVertical => renderer.isVertical;
+
+  Size get viewportSize => renderer.size;
+
+  double get contentExtent => _contentExtent;
+  double _contentExtent = 0.0;
+  void set contentExtent (double value) {
+    assert(value != null);
+    assert(value >= 0.0);
+    if (_contentExtent == value)
+      return;
+    _contentExtent = value;
+    renderer?.markNeedsPaint();
+  }
+
+  double get scrollOffset => _scrollOffset;
+  double _scrollOffset = 0.0;
+  void set scrollOffset (double value) {
+    assert(value != null);
+    assert(value >= 0.0 && value <= 1.0);
+    if (_scrollOffset == value)
+      return;
+    _scrollOffset = value;
+    renderer?.markNeedsPaint();
+  }
+
+  Future scrollStarted() => new Future.value();
```

Comment:
Not sure which methods you're referring to.  The scrollOffset r/w property, which exists so the painter can draw something that depends on the scrollOffset? Or the scrollStarted(), scrollEnded() methods, which subclasses can override to start animations when scrolling starts or ends?